### ISSUE TITLE
Map int from API call to built_value Enum

### ISSUE THREAD

**Issue by SandroMaglione** [google/built_value.dart#815]
I have an API that stores a value as an integer. Those value, therefore, comes in JSON format like this:
```json
{
   "language": 1
}
```

I then need to map this integer to an enum inside the application. The problem I have is that, with built_value, enums are mapped as Strings. Therefore, if for example I create an enum like this, the conversion would be from `Language.english` to `english` instead of `1`:
```dart
part 'language.g.dart';

class Language extends EnumClass {
  static const Language english = _$english;
  static const Language italian = _$italian;

  const Language._(String name) : super(name);

  static BuiltSet<Language> get values => _$values;
  static Language valueOf(String name) => _$valueOf(name);
  static Serializer<Language> get serializer => _$languageSerializer;
}
```
I would like to convert the integer from JSON to `Language.english`, and then be able to convert it back to an integer, mapping the enum values to the correct integer.

I was using json_serializable, and I just needed to add the JsonKey annotation to implement this:
```dart
enum Language {
  @JsonValue(1)
  english,

  @JsonValue(2)
  italian,
}
```

Is there an easy way or annotation to implement this with built_value?

**Comment by davidmorgan** [comment]
This is not yet supported; you can set any string for the wire value, but not an integer. There's an existing feature request for this but I couldn't find it.

Anyway, yes, it makes sense. Will look at adding at some point. Thanks!

**Comment by SandroMaglione** [comment]
It is probably related to this issue
[Support numbers in @BuiltValueEnumConst #657](https://github.com/google/built_value.dart/issues/657)

**Comment by davidmorgan** [comment]
That's the one. Thanks!

**Comment by SandroMaglione** [comment]
Implemented in #852
